imapclient/imapclient.py

```diff
@@ -1505,6 +1505,17 @@ def expunge(self, messages=None):
         tag = self._imap._command("EXPUNGE")
         return self._consume_until_tagged_response(tag, "EXPUNGE")
 
+    @require_capability("UIDPLUS")
+    def uid_expunge(self, messages):
+        """Same functionality as ``expunge``, but *messages* must be
+        specified, and the capability UIDPLUS is tested beforehand. It should
+        be more fail-proof than ``expunge`` with *messages*, which cannot be
+        updated to prevent breaking compability with existing codebase.
```

Comment:
I don't think the docstring requires quite as much detail. Something like this should do:

>  Expunge deleted messages with the specified message ids from the folder. This requires the UIDPLUS capability.
>
> See :rfc:`4315#section-2.1` section 2.1 for more details.

 Then please extend the docstring on the expunge method to discourage use of `messages` there. Something like:

> Use of the *messages* argument is discouraged. Please see the uid_expunge method instead.

I would be ok with this replacing the existing paragraph about the `messages` argument.

Thanks!

Reply:
@mjs 
Your choice of words is better than mine, I am using it without modification. New commit e0cbf56 and ea5ef55